pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -2154,6 +2157,15 @@ type OSSArtifact struct {
 	Key string `json:"key" protobuf:"bytes,2,opt,name=key"`
 }
 
+// LifecycleRule specifies how to manage bucket's lifecycle
+type LifecycleRule struct {
+	// MarkInfrequentAccessAfterDays is the number of days before we convert the objects in the bucket to Infrequent Access (IA) storage type
+	MarkInfrequentAccessAfterDays *int32 `json:"markInfrequentAccessAfterDays,omitempty" protobuf:"bytes,1,opt,name=markInfrequentAccessAfterDays"`
```

Comment:
why use pointer type?

Reply:
I was trying to make this consistent with other types in workflow.

Changed to int32.